raiwidgets/raiwidgets/_cohort.py

```diff
@@ -0,0 +1,186 @@
+        :type method: str
+        :param arg: List of values to be used by the cohort filter.
+        :type arg: list
+        :param column: The column name from the dataset on which the filter
+                       will be applied.
+        :type column: str
+        """
+        if method not in CohortFilterMethods.ALL:
+            raise UserConfigValidationException(
+                "Got unexpected value {0} for method. "
+                "Expected either of {1}.".format(
+                    method, " or ".join(CohortFilterMethods.ALL))
+            )
+        if not isinstance(column, str):
+            raise UserConfigValidationException(
+                "Got unexpected type {0} for column. "
+                "Expected string type.".format(type(column))
+            )
+        if not isinstance(arg, list):
+            raise UserConfigValidationException(
+                "Got unexpected type {0} for arg. "
+                "Expected list type.".format(type(arg))
+            )
+        if len(arg) == 0:
+            raise UserConfigValidationException(
+                "Empty list supplied for arg."
+            )
+
+        if method in CohortFilterMethods.SINGLE_VALUE_METHODS:
+            if len(arg) != 1:
+                raise UserConfigValidationException(
+                    "Expected a single value in arg "
+                    "for cohort methods {0}.".format(
+                        " or ".join(CohortFilterMethods.SINGLE_VALUE_METHODS))
+                )
+            if not isinstance(arg[0], int) and not isinstance(arg[0], float):
+                raise UserConfigValidationException(
+                    "Expected int or float type for "
+                    "arg with cohort methods {0}.".format(
+                        " or ".join(CohortFilterMethods.SINGLE_VALUE_METHODS))
+                )
+
+        if method == CohortFilterMethods.InTheRangeOf:
+            if len(arg) != 2:
+                raise UserConfigValidationException(
+                    "Expected two entries in arg for "
+                    "cohort method {0}.".format(
+                        CohortFilterMethods.InTheRangeOf)
+                )
+            if ((not isinstance(arg[0], int) and
+                 not isinstance(arg[0], float)) or
+                (not isinstance(arg[1], int) and
+                 not isinstance(arg[1], float))):
+                raise UserConfigValidationException(
+                    "Expected int or float type for arg "
+                    "with cohort method {0}.".format(
+                        CohortFilterMethods.InTheRangeOf)
+                )
+
+
+class Cohort:
+    """Defines the cohort which will be injected from SDK into the Dashboard.
+
+    :param method: Name of the cohort.
+    :type method: str
+    """
+    def __init__(self, name: str):
+        """Defines the cohort which will be injected from SDK into the Dashboard.
+
+        :param method: Name of the cohort.
```

Comment:
is it "method" or "name" for the param here?

```
def __init__(self, name: str):
...
    :param method: Name of the cohort.
```

Reply:
Changed it to name in doc strings.